apps/gitness/src/pages/create-workspace.tsx

```diff
@@ -0,0 +1,61 @@
+import {
+  useCreateSpaceMutation,
+  OpenapiCreateSpaceRequest,
+  CreateSpaceErrorResponse,
+  CreateSpaceOkResponse
+} from '@harnessio/code-service-client'
+import { CreateWorkspacePage } from '@harnessio/playground'
+import { useNavigate } from 'react-router-dom'
+import { useAppContext } from '../framework/context/AppContext'
+
+export default function CreateWorkspace() {
+  const navigate = useNavigate()
+  const { addSpace } = useAppContext() // Get the spaces and addSpace function from context
+
+  //Todo: think about using throw error instead of alert for improvement
+  const handleCreateSpaceError = (error: CreateSpaceErrorResponse) => {
+    if (error.message) {
+      console.error('API Error:', error.message)
+      alert(`Error: ${error.message}`)
```

Comment:
Lets not add window alert here as it might cause some annoyance to the user. Let's remove console logging as well. Instead we should just render an error below the `Create Workspace` button as well.

Reply:
pass error message down to components in playground
![Screenshot 2024-09-18 at 11 50 00 AM](https://github.com/user-attachments/assets/3c76ebf4-b3f5-489e-9624-96d52e7cb79a)
![Screenshot 2024-09-18 at 12 15 18 PM](https://github.com/user-attachments/assets/530d5124-fc2a-49b1-994b-0881917a77e6)
![Screenshot 2024-09-18 at 12 15 36 PM](https://github.com/user-attachments/assets/ec3c4841-0aad-4b26-a395-d8056946b73b)